### ISSUE TITLE
Bounced messages have no body

### ISSUE THREAD

**Issue by behrang** [ton-community/sandbox#17]
When a message is bounced from a smart contract, it doesn't have any body. I expect to see 0xFFFFFFFF plus the first 256 bits of the original body as specified in:
https://ton.org/docs/develop/smart-contracts/guidelines/processing
or
https://ton.org/docs/develop/smart-contracts/guidelines/non-bouncable-messages

Currently I see something like this in the bounced message:

```
      {
        from: EQBnBlCAuvnW5Vwb_msWN24xHQM68DO5kFSpmzV7rUDcmISC,
        to: Ef-fE9ukteSTIxM1y2DdcW2Thlq2kVwb6HSY0ketaGqGqlub,
        value: 901775000n,
        body: x{},
        initData: undefined,
        initCode: undefined,
        deploy: false,
        lt: 8000000n,
        now: 1676979972,
        outMessagesCount: 0,
        oldStatus: 'active',
        endStatus: 'active',
        totalFees: 17700000n,
        aborted: true,
        destroyed: false,
        exitCode: 9,
        success: false
      }
```

The reason for exit code 9 here is that I was trying to read from the body, which was unexpectedly empty.

**Comment by krigga** [comment]
I'm not 100% sure, but I think if you're seeing exit code 9 then it's not the bounced message. Does the input message have `bounce` flag set?

You can also just share the code for me to see if that's easier

**Comment by krigga** [comment]
Oh wait I see what you mean now, but the code eill be helpful regardless

**Comment by krigga** [comment]
I've confirmed the bug locally, will investigate

**Comment by krigga** [comment]
I believe I've found the underlying issue, it's in the C++ code of the emulator. I will make a PR to the main TON repo to fix that, but I'll probably include a forked and fixed version in the next release of sandbox.

**Comment by behrang** [comment]
At last, I created a valid issue.

Great. This was a problem in a few of my test cases. For now, I'll comment them out, and wait for the next version.

The case that I'm testing is the the `burn` case in jettons. I want to test the case that when the child burns some tokens, if the parent throws an exception and fails, the burned tokens are minted back. So I needed to simulate the failure in my test cases. I can send the code if you want it.

**Comment by krigga** [comment]
> At last, I created a valid issue.

No worries, your issues will become better docs

**Comment by krigga** [comment]
Here's the PR to main TON repo https://github.com/ton-blockchain/ton/pull/620

The forked fix is linked above and will be present in the next release

**Comment by krigga** [comment]
Fix released in v0.5.0

**Comment by behrang** [comment]
Hey thanks, I upgraded and I can confirm it now works as expected.
